wagtailio/packages/viewsets.py

```diff
@@ -0,0 +1,39 @@
+from wagtail.admin.ui.tables import BooleanColumn
+from wagtail.admin.viewsets.model import ModelViewSet, ModelViewSetGroup
+
+from .models import Grid, Package
+
+
+class BaseModelViewSet(ModelViewSet):
+    icon = "doc-full-inverse"
+    list_display = [
+        "title",
+        BooleanColumn(
+            "publish",
+            sort_key="publish",
+        ),
+    ]
+    list_filter = ["publish"]
+    search_fields = ["title"]
+
+    def on_register(self):
+        super().on_register()
+        # used in `get_admin_url` for the models
+        self.model.model_viewset = self
+
+
+class GridAdmin(BaseModelViewSet):
+    model = Grid
+    form_fields = ["publish", "title", "slug", "description"]
+
+
+class PackageAdmin(BaseModelViewSet):
+    model = Package
+    form_fields = ["publish", "title", "slug", "repo_description"]
+
+
+class PackagesAdminGroup(ModelViewSetGroup):
+    menu_label = "Package"
+    menu_icon = "folder-open-inverse"
```

Comment:
Nitpick: We're using the same iconography as the page(s) button, as these are landmarks on a page it feels a little strange to me? 

Perhaps we could use `form` or `tasks`? [options we could use](https://docs.wagtail.org/en/latest/advanced_topics/icons.html)

Reply:
changed to https://fontawesome.com/icons/cube?f=classic&s=solid